.ibm/images/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17
+FROM golang:1.18
```

Comment:
@dharmit Just for double-checking (as I am not sure when this image is being built and pushed for our CI), I just tried to build this image, and the `go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045` command did not pass:

```
$ podman image build -t redhat-developer/odo-ibm-pipeline .ibm/images

...
STEP 4/5: USER tester
--> 1c5c6b2312b
STEP 5/5: RUN go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045 &&     ibmcloud plugin install -f cloud-object-storage &&     ibmcloud plugin install -f kubernetes-service
go: go.mod file not found in current directory or any parent directory.
        'go get' is no longer supported outside a module.
        To build and install a command, use 'go install' with a version,
        like 'go install example.com/cmd@latest'
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
Error: error building at STEP "RUN go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045 &&     ibmcloud plugin install -f cloud-object-storage &&     ibmcloud plugin install -f kubernetes-service": error while running runtime: exit status 1

```

Reply:
After a quick chat with Philippe, that is the image we use in the configuration of the stages for our IBM pipelines. And it is currently built and pushed manually to `quay.io/odo-dev/ibmcloudtest-image:latest`.
The container spawned by IBM Cloud to run our tests is then created off of this image.

BTW, the image could build successfully using `golang:1.17` as base image, but with a warning:
```
STEP 4/5: USER tester
--> e9a2883e72f
STEP 5/5: RUN go get github.com/kadel/odo-robot@965ea0dd848856691bfc76e6824a8b787b950045 &&     ibmcloud plugin install -f cloud-object-storage &&     ibmcloud plugin install -f kubernetes-service
go: downloading github.com/kadel/odo-robot v0.0.0-20211217142858-965ea0dd8488
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
go: downloading github.com/google/go-github/v41 v41.0.0
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
go: downloading google.golang.org/appengine v1.6.7
go: downloading golang.org/x/crypto v0.0.0-20210817164053-32db794688a5
go: downloading github.com/google/go-querystring v1.1.0
go: downloading github.com/golang/protobuf v1.4.2
go: downloading google.golang.org/protobuf v1.25.0
go get: installing executables with 'go get' in module mode is deprecated.
        Use 'go install pkg@version' instead.
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
Looking up 'cloud-object-storage' from repository 'IBM Cloud'...
Plug-in 'cloud-object-storage 1.6.0' found in repository 'IBM Cloud'
Attempting to download the binary file...
 16.06 MiB / 16.06 MiB  100.00% 1s01s
16842752 bytes downloaded
Installing binary...
OK
```

So I guess we need to install using `go install` instead..

Reply:
This image is used for CI Pipelines on IBM Cloud.
Effectively, `go get` on this context does not work anymore with go 1.18, you will need to replace it with `go install`

Reply:
> So I guess we need to install `odo-robot` using `go install` instead..

> Effectively, go get on this context does not work anymore with go 1.18, you will need to replace it with go install

I should have read the error message closely. I only replied based on the part about "where the image is built" in @rm3l's initial comment. 

I have seen this error frequently since moving to go 1.18 locally and have pushed a fix.

Thanks @rm3l & @feloy